File: iOSApp/TVMovieAppIOS/TVMovieAppIOS/ui/views/ListView.swift
import SwiftUI

enum Route: Hashable {
    case movieDetail(String)
}


struct ListView: View {
    @State private var path: [Route] = []

    var body: some View {
        NavigationStack(path: $path) {
            AppBackground {
                ScrollView {
                    VStack(alignment: .leading, spacing: 28) {

                        HighlightHero(feedId: FeedIds.mostSeen.rawValue,
                            onSelect: { movie in
                                path.append(.movieDetail(movie.urlPath))
                            }
                        )
                        
                        FeedSection(
                            title: "For deg",
                            feedId: FeedIds.recommended.rawValue,
                            variant: "poster",
                            onSelect: { movie in
                                path.append(.movieDetail(movie.urlPath))
                            }
                        )

                        FeedSection(
                            title: "Vist på festival",
                            feedId: FeedIds.festival.rawValue,
                            variant: "landscape",
                            onSelect: { movie in
                                path.append(.movieDetail(movie.urlPath))
                            }
                        )
                        FeedSection(
                            title: "Alltid film",
                            feedId: FeedIds.focus.rawValue,
                            variant: "landscape",
                            onSelect: { movie in
                                path.append(.movieDetail(movie.urlPath))
                            }
                        )

                        FeedSection(
                            title: "Nyankommende",
                            feedId: FeedIds.newArrivals.rawValue,
                            variant: "landscape",
                            onSelect: { movie in
                                path.append(.movieDetail(movie.urlPath))
                            }
                        )
                        FeedSection(
                            title: "Kjøp eller lei",
                            feedId: FeedIds.buyRent.rawValue,
                            variant: "poster",
                            onSelect: { movie in
                                path.append(.movieDetail(movie.urlPath))
                            }
                        )

                    }
                    .padding(.top, 20)
                }
            }
            .navigationDestination(for: Route.self) { route in
                switch route {
                case .movieDetail(let urlPath):
                    DetailsView(urlPath: urlPath)
                }
            }
        }
    }
}

File: AndroidApp/app/src/main/java/com/tv2oppgave/tvmovieapp/ui/ListView/ListView.kt
package com.tv2oppgave.tvmovieapp.ui.ListView

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import com.tv2oppgave.tvmovieapp.data.models.FeedIds

@Composable
fun ListView(
    navController: NavController
) {
    LazyColumn(
        verticalArrangement = Arrangement.spacedBy(24.dp),
    ) {
        item {
            HighlightHero(
                navController = navController,
                feedId = FeedIds.MostSeen,
            )
        }
        item { FeedView(navController = navController, FeedIds.Recommended, "For deg", "landscape") }
        item { FeedView(navController = navController, FeedIds.Festival, "Vist på festival", "poster") }
        item { FeedView(navController = navController, FeedIds.Focus, "Alltid film", "landscape") }
        item { FeedView(navController = navController, FeedIds.NewArrivals, "Nyeankommede", "landscape") }
        item { FeedView(navController = navController, FeedIds.BuyRent, "Kjøp eller lei", "poster") }
    }
}